server/controllers/authentication/authentication.spec.js

```diff
@@ -0,0 +1,218 @@
+import supertest from 'supertest';
+import sinon from 'sinon';
+import sgMail from '@sendgrid/mail';
+import model from '<models>';
+import app from '<server>/app';
+import { getData, inputObject } from '<fixtures>/data';
+import { signup, verifyAccount, login } from '<controllers>/authentication';
+import { tokenGenerator } from '<helpers>/utils';
+
+const { User, sequelize } = model;
+
+const server = () => supertest(app);
+
+let mailSender, setApiKey, user, token;
+
+const { TOKEN_EXPIRY_DATE, SECRET } = process.env;
+
+beforeAll(async () => {
+  sequelize.sync();
+
+  const userData = getData({ email: 'mosinmiloluwa.o@gmail.com' });
+  const newUser = await User.create(userData);
+  user = newUser.get({ plain: true });
+
+  token = tokenGenerator(user.id, false, TOKEN_EXPIRY_DATE, SECRET);
+  setApiKey = sinon.stub(sgMail, 'setApiKey');
+  mailSender = sinon.stub(sgMail, 'send').returns(Promise.resolve({}));
```

Comment:
Nice one with the stub. How about we make use of `jest`, since it provides mocking out of the box. See the suggestion below:

```suggestion
  jest.spyOn(sgMail, 'setApiKey'). mockReturnValue();
  jest.spyOn(sgMail, 'send'). mockResolvedValue({});
```